objects/src/assets/mod.rs

```diff
@@ -37,12 +37,32 @@ use core::{fmt, ops::Deref};
 ///   This guarantees that finding two assets with the same hash is infeasible.
 /// - Lastly, the first bit of the result is set to ZERO, and the last 32 bits of the 3rd element
 ///   are set to 2^31 (i.e., ONE followed by 32 ZEROs).
-#[derive(Debug, Clone, Eq, PartialEq)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub enum Asset {
     Fungible(FungibleAsset),
     NonFungible(NonFungibleAsset),
 }
 
+impl Asset {
+    /// Returns true if this asset is the same as the specified asset.
+    ///
+    /// Two assets are defined to be the same if:
+    /// - For fungible assets, if they were issued by the same faucet.
+    /// - For non-fungible assets, if the assets are identical.
+    pub fn is_same(&self, other: &Self) -> bool {
+        match self {
+            Asset::Fungible(self_asset) => match other {
+                Asset::Fungible(other_asset) => self_asset.is_same(other_asset),
+                Asset::NonFungible(_) => false,
+            },
+            Asset::NonFungible(self_asset) => match other {
+                Asset::Fungible(_) => false,
+                Asset::NonFungible(other_asset) => self_asset == other_asset,
+            },
+        }
+    }
```

Comment:
That can be simplified into a match of tuples

```suggestion
    pub fn is_same(&self, other: &Self) -> bool {
        use Asset::*;
        match (self, other) {
            (Fungible(l), Fungible(r)) => l.is_same(r),
            (NonFungible(l), NonFungible(r)) => l == r,
            _ => false,
        }
    }
```

Reply:
Yes, much better! Thank you!